src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -135,6 +136,24 @@ void Instance::Shutdown()
     mpBaseDriver->Shutdown();
 }
 
+#if !CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION
+void Instance::SendNonConcurrentConnectNetworkResponse()
+{
+    auto commandHandleRef = std::move(mAsyncCommandHandle);
+    auto commandHandle    = commandHandleRef.Get();
+    if (commandHandle == nullptr)
+    {
+        return;
+    }
+
+    DeviceLayer::ConnectivityMgr().GetBleLayer()->mState = Ble::BleLayer::kState_Disconnecting;
```

Comment:
Why are we directly touching BLE here? We should NOT be touching internal state of BLE layer from this place, and at the very least, if done, it should not be via member fields.

Reply:
I looked at sending an event to the Commissioning Manager which would call the BLE, but that has race conditions. The disconnecting state needs setting prior to the sending of the BTP packet containing the ConnectNetworkResponse(Success) but the event is queued at the same time AddResponse() is called. Which means there is the potential that the BTP packet completes prior to the event being handled. Therefore I have created a function which is now called to set the mState within the BLE layer rather than directly accessing the member field.